app/grandchallenge/invoices/tasks.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.db import transaction
+from django.utils.timezone import now
+
+from grandchallenge.core.celery import acks_late_micro_short_task
+from grandchallenge.invoices.emails import (
+    send_challenge_invoice_issued_notification,
+    send_challenge_outstanding_invoice_reminder,
+)
+
+
+@acks_late_micro_short_task
+@transaction.atomic
+def send_challenge_outstanding_invoice_reminder_emails():
+    from grandchallenge.invoices.models import Invoice
+
+    _now = now()
+    outstanding_invoices = Invoice.objects.filter(
+        payment_type__in=[
+            Invoice.PaymentTypeChoices.PREPAID,
+            Invoice.PaymentTypeChoices.POSTPAID,
+        ],
+        payment_status=Invoice.PaymentStatusChoices.ISSUED,
+        issued_on__lt=_now - settings.CHALLENGE_INVOICE_OUTSTANDING_CUTOFF,
+    )
+    for invoice in outstanding_invoices:
+        send_challenge_outstanding_invoice_reminder(invoice)
+
+
+@acks_late_micro_short_task
+@transaction.atomic
+def send_challenge_invoice_issued_notification_emails(*, pk):
+    from grandchallenge.invoices.models import Invoice
+
+    invoice = Invoice.objects.get(pk=pk)
+    send_challenge_invoice_issued_notification(invoice)
```

Comment:
Does this need to be done asynchronously? It's just sending a handful of emails at most, right?

Reply:
This is to decouple the saving and email sending as sometime these take a while and you can get lock contention (for instance with the notification email creation on forum posts).